src/maintenance/azext_maintenance/generated/_params.py

```diff
@@ -0,0 +1,210 @@
+        c.argument('resource_type', type=str, help='Resource type')
+        c.argument('resource_name', type=str, help='Resource identifier')
+        c.argument('configuration_assignment_name', options_list=['--name', '-n', '--configuration-assignment-name'],
+                   type=str, help='Configuration assignment name')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('maintenance_configuration_id', type=str, help='The maintenance configuration Id')
+        c.argument('resource_id', type=str, help='The unique resourceId')
+
+    with self.argument_context('maintenance assignment update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('provider_name', type=str, help='Resource provider name')
+        c.argument('resource_type', type=str, help='Resource type')
+        c.argument('resource_name', type=str, help='Resource identifier')
+        c.argument('configuration_assignment_name', options_list=['--name', '-n', '--configuration-assignment-name'],
+                   type=str, help='Configuration assignment name')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('maintenance_configuration_id', type=str, help='The maintenance configuration Id')
+        c.argument('resource_id', type=str, help='The unique resourceId')
+
+    with self.argument_context('maintenance assignment delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('provider_name', type=str, help='Resource provider name')
+        c.argument('resource_parent_type', type=str, help='Resource parent type')
+        c.argument('resource_parent_name', type=str, help='Resource parent identifier')
+        c.argument('resource_type', type=str, help='Resource type')
+        c.argument('resource_name', type=str, help='Resource identifier')
+        c.argument('configuration_assignment_name', options_list=['--name', '-n', '--configuration-assignment-name'],
+                   type=str, help='Unique configuration assignment name')
+
+    with self.argument_context('maintenance assignment list-parent') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('provider_name', type=str, help='Resource provider name')
+        c.argument('resource_parent_type', type=str, help='Resource parent type')
+        c.argument('resource_parent_name', type=str, help='Resource parent identifier')
+        c.argument('resource_type', type=str, help='Resource type')
+        c.argument('resource_name', type=str, help='Resource identifier')
+
+    with self.argument_context('maintenance configuration show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
```

Comment:
No need to register "resource_group_name" as  it will be registered as global arguments automatically.